src/test/java/selenium/pageobjects/WebLabAnswerPage.java

```diff
@@ -0,0 +1,40 @@
+package selenium.pageobjects;
+
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+public class WebLabAnswerPage extends BasePageObject {
+
+    public WebLabAnswerPage(WebDriver driver, String url) {
+        super(driver, url);
+    }
+
+    @FindBy(xpath = "/html/body/div[2]/div[5]/div/div/div[3]/div[1]/div/div/div[1]")
+    private WebElement solutionDiv;
+
+    @FindBy(xpath = "/html/body/div[2]/div[5]/div/div/div[3]/div[1]/div/div/a[1]")
+    private WebElement unlockAnswerButton;
+
+    public String getSolution() {
+        this.awaitPageLoaded();
+
+        // unlock answer if it is not yet unlocked
+        if (this.isDisplayed(unlockAnswerButton)) {
+            this.unlockAnswerButton.click();
+            this.awaitElementVisibility(this.solutionDiv);
+        }
+
+        String script = "return document.getElementById('solution').value;";
```

Comment:
You can't get the element via selenium?

Reply:
This element is invisible, so I don't think it's possible to get it directly via selenium.

Reply:
https://stackoverflow.com/questions/20012442/using-selenium-can-i-get-the-text-of-an-invisible-element#24336401

Does this help?

Reply:
Thank you, this did the job. It seems like `getAttribute` works even if the element is not displayed.